components/infra-proxy-service/migrations/pipeline/utility.go

```diff
@@ -0,0 +1,68 @@
+package pipeline
+
+import (
+	"context"
+
+	"github.com/chef/automate/components/infra-proxy-service/storage"
+	log "github.com/sirupsen/logrus"
+)
+
+// StoreOrgs reads the Result struct and populate the orgs table
+func StoreOrgs(ctx context.Context, st storage.Storage, mst storage.MigrationStorage, res Result) (Result, error) {
+	var err error
+	var msg string
+	var totalSucceeded, totalSkipped, totalFailed int64
+	_, err = mst.StartOrgMigration(ctx, res.Meta.MigrationID, res.Meta.ServerID)
+	if err != nil {
+		return res, err
+	}
+	log.Info("Starting the organisation migration phase for migration id: ", res.Meta.MigrationID)
+	for _, org := range res.ParsedResult.Orgs {
+		err, _ = StoreOrg(ctx, st, org, res.Meta.ServerID)
+		if err != nil {
+			msg = err.Error()
+			totalFailed++
+			break
+		}
+		if org.ActionOps == Skip {
+			totalSkipped++
+			continue
+		}
+		totalSucceeded++
+	}
+	if err != nil {
+		log.Errorf("Failed to migrate orgs for migration id %s : %s", res.Meta.MigrationID, err.Error())
```

Comment:
If there is a failure, irrespective of whether the DB update for status was successful, we should return the error